google-compute-engine/src/main/java/org/jclouds/googlecomputeengine/compute/strategy/CreateNodesWithGroupEncodedIntoNameThenAddToSet.java

```diff
@@ -103,14 +103,15 @@
 
       String sharedResourceName = namingConvention.create().sharedNameForGroup(group);
       Template mutableTemplate = template.clone();
-      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class
-            .cast(mutableTemplate.getOptions());
+      GoogleComputeEngineTemplateOptions templateOptions = GoogleComputeEngineTemplateOptions.class.cast(mutableTemplate.getOptions());
       assert template.getOptions().equals(templateOptions) : "options didn't clone properly";
 
-      // get or insert the network and insert a firewall with the users
-      // configuration
+      // get or insert the network and insert a firewall with the users configuration
       Network network = getOrCreateNetwork(templateOptions, sharedResourceName);
-      getOrCreateFirewalls(templateOptions, network, firewallTagNamingConvention.get(group));
+      // add firewalls *only* if the template options don't contain explicit network
+      if (templateOptions.network() == null) {
```

Comment:
Problem with GCE is that it creates a firewall in a network (`createInNetwork`): security groups are not modeled, so the wrapping `container` is the Network.
Anyway, thanks for your thoughts. I'll come up with a different patch.